File: src/main/java/dev/ambryn/task/models/Finished.java
package dev.ambryn.task.models;

import dev.ambryn.task.interfaces.State;

public class Finished extends State {

    public Finished(Task context) {
        this.context = context;
        this.status = Status.FINISHED;
    }

    @Override
    public void start() {
        return;
    }

    @Override
    public void error() {
        return;
    }

    @Override
    public void finish() {
        return;
    }

    @Override
    public void cancel() {
        return;
    }
}

File: src/main/java/dev/ambryn/task/models/Finished.kt
package dev.ambryn.task.models

import dev.ambryn.task.interfaces.State

class Finished(context: Task) : State(context) {
    override var status = Status.FINISHED

    override fun start() {
        return
    }

    override fun error() {
        return
    }

    override fun finish() {
        return
    }

    override fun cancel() {
        return
    }
}
